sentry-spring/src/test/kotlin/io/sentry/spring/reactive/SentryReactiveWebFilterTest.kt

```diff
@@ -0,0 +1,62 @@
+package io.sentry.spring.reactive
+
+import com.nhaarman.mockitokotlin2.check
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.sentry.Breadcrumb
+import io.sentry.IHub
+import io.sentry.SentryOptions
+import javax.servlet.ServletRequestEvent
+import kotlin.test.Test
+import org.assertj.core.api.Assertions
+import org.springframework.mock.http.server.reactive.MockServerHttpRequest
+import org.springframework.mock.web.server.MockServerWebExchange
+import org.springframework.web.server.WebFilterChain
+
+class SentryReactiveWebFilterTest {
+
+    private class Fixture {
+        val baseHub = mock<IHub>()
+        val hub = mock<IHub>()
+        val filter = SentryReactiveWebFilter(baseHub, SentryOptions())
+        val request = MockServerHttpRequest
+            .post("http://localhost:8080/some-uri")
+            .build()
+        val event = mock<ServletRequestEvent>()
+        val webExchange = MockServerWebExchange.from(request)
+        val chain = mock<WebFilterChain>()
+
+        init {
+            whenever(baseHub.clone()).thenReturn(hub)
+        }
+    }
+
+    private val fixture = Fixture()
+
+    @Test
+    fun `clones hub when request arrives`() {
+        fixture.filter.filter(fixture.webExchange, fixture.chain)
+
+        verify(fixture.baseHub).clone()
+        verify(fixture.hub).pushScope()
+    }
+
+    @Test
+    fun `adds breadcrumb when request gets initialized`() {
+        fixture.filter.filter(fixture.webExchange, fixture.chain)
+
+        verify(fixture.hub).addBreadcrumb(check { it: Breadcrumb ->
+            Assertions.assertThat(it.getData("url")).isEqualTo("/some-uri")
+            Assertions.assertThat(it.getData("method")).isEqualTo("POST")
+            Assertions.assertThat(it.type).isEqualTo("http")
```

Comment:
same line of https://github.com/getsentry/sentry-java/pull/999/files#r507518657

Reply:
@marandaneto can you clarify just a little please? The link does not show me any specific file.

Reply:
oh sorry, when that comment was solved, the anchor is broken, I do this all the time.
its pretty much the same as the `Mockito.never()` issue, using `com.nhaarman.mockitokotlin2` for testing, you dont need to use the static methods from `Assertions`, you can use `assertEquals` directly as all the other tests

Reply:
@marandaneto Thanks for taking the time to clarify this, it's much clearer now.